tests/integration/ha/test_horizontal_scaling.py

```diff
@@ -54,6 +57,11 @@ async def c_writes_runner(ops_test: OpsTest, c_writes: ContinuousWrites):
 @pytest.mark.skip_if_deployed
 async def test_build_and_deploy(ops_test: OpsTest) -> None:
     """Build and deploy one unit of OpenSearch."""
+    # it is possible for users to provide their own cluster for HA testing.
+    # Hence, check if there is a pre-existing cluster.
+    if await app_name(ops_test):
+        return
+
     my_charm = await ops_test.build_charm(".")
     await ops_test.model.set_config(MODEL_CONFIG)
```

Comment:
question: should this model config be applied to a user provided cluster?

Reply:
I think we should assume that a user provided cluster has been pre-configured / tuned by the user, so we shouldn't intrude in the model configuration